### PR TITLE
Get comments by article

### PR DESCRIPTION
**Pull request by nickWoott**
Completed the get comments by article endpoint. 
All logic done in SQL query.

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -166,13 +166,65 @@ describe("GET/api/users", () => {
...
-//       });
-//   });
-// });
+describe("/api/articles", () => {
+  test("200, response with an array", () => {
+    return request(app)
+      .get("/api/articles")
+      .expect(200)
+      .then((res) => {
+        expect(res.body).toBeInstanceOf(Array);
```
Comment (dentednerd) [inline __tests__/app.test.js:175]: How might you go about testing the contents of this array?

File: __tests__/app.test.js [hunk 2]
```diff
@@ -166,13 +166,65 @@ describe("GET/api/users", () => {
...
+  test("200: returns an array", () => {
+    return request(app)
+      .get("/api/articles/1/comments")
+      .expect(200)
+      .then((res) => {
+        expect(res.body).toBeInstanceOf(Array);
+      });
+  });
+  test("200: returns an array with correct properties", () => {
+    return request(app)
```
Comment (dentednerd) [inline __tests__/app.test.js:207]: You'd be able to combine both of these tests into one

File: controllers/articles.controllers.js [hunk 3]
```diff
@@ -32,12 +33,23 @@ exports.increaseVotes = (req, res, next) => {
...
+    .then((articles) => {
+      res.status(200).send(articles);
+    })
+    .catch((err) => {
+      next(err);
+    });
+};
+
+exports.getComments = (req, res, next) => {
+  const articleId = req.params.article_id;
```
Comment (dentednerd) [inline controllers/articles.controllers.js:47]: Use deconstruction here instead!

File: __tests__/app.test.js (hunk 1 above)
Comment (nickWoott) [reply __tests__/app.test.js:175]: I will do a forEach and a toMatchObject.

File: models/articles.models.js [hunk 4]
```diff
@@ -33,4 +34,32 @@ exports.updateVotes = async (articleId, update) => {
   return results.rows[0];
 };
 
-// exports.selectArticles;
+exports.selectArticles = async () => {
+  const articles = await db.query("SELECT * FROM articles");
+  const comments = await db.query("SELECT * FROM comments");
+  let articleArray = [];
+  articles.rows.forEach((article) => {
+    let commentCount = 0;
```
Comment (dentednerd) [inline models/articles.models.js:42]: Don't create your comment count this way! Use SQL

File: models/articles.models.js [hunk 5]
```diff
@@ -33,4 +34,32 @@ exports.updateVotes = async (articleId, update) => {
   return results.rows[0];
 };
 
-// exports.selectArticles;
+exports.selectArticles = async () => {
+  const articles = await db.query("SELECT * FROM articles");
+  const comments = await db.query("SELECT * FROM comments");
```
Comment (dentednerd) [inline models/articles.models.js:39]: You don't need to make two separate SQL queries to get what you want here.

File: utilities/utilities.js [hunk 6]
```diff
@@ -0,0 +1,23 @@
+// const db = require("../db/connection");
```
Comment (dentednerd) [inline utilities/utilities.js:1]: This file has nothing but comments in it... did you really need to commit it?

File: utilities/utitilities.test.js [hunk 7]
```diff
@@ -0,0 +1,30 @@
+const {
+  addCommentCount,
+  returnUpdatedArticle,
+} = require("../utilities/utilities.js");
+const db = require("../db/connection");
+
+describe.skip("addCommentCount", () => {
```
Comment (dentednerd) [inline utilities/utitilities.test.js:7]: This is not a good test. Please have a rethink.

File: __tests__/app.test.js [hunk 8]
```diff
@@ -166,13 +166,65 @@ describe("GET/api/users", () => {
...
+describe("/api/articles", () => {
+  test("200, response with an array", () => {
+    return request(app)
+      .get("/api/articles")
+      .expect(200)
+      .then((res) => {
+        expect(res.body).toBeInstanceOf(Array);
+      });
+  });
+  test("200: responds with updated comments", () => {
```
Comment (dentednerd) [inline __tests__/app.test.js:178]: Your choice of words here indicates that you're testing comments that have been updated.

**Review by dentednerd — CHANGES_REQUESTED** [review]
This needs some refactoring, Nick. Feel free to nchelp if you need advice.

**Pull request merged**